### PR TITLE
iio-oscilloscope: init at 0.17

### PR DESCRIPTION
**Pull request by wucke13**
## Description of changes

This is an oscilloscope & spectrum analyzer application for Analog Device's Industrial IO (IIO) stuff, like the Adalm PlutoSDR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by FliegendeWurst** [comment]
Another PR for the same package: #280521

File: pkgs/by-name/ii/iio-oscilloscope/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{ lib
+, stdenv
```
Comment (FliegendeWurst) [inline pkgs/by-name/ii/iio-oscilloscope/package.nix:2]: Please format with the new formatter.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ii/iio-oscilloscope/package.nix (hunk 1 above)
Comment (wucke13) [reply pkgs/by-name/ii/iio-oscilloscope/package.nix]: Thank you for the hint, @FliegendeWurst . Adjusted accordingly, also rebased to current master.

File: pkgs/by-name/ii/iio-oscilloscope/package.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    pcre2
+  ];
+
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=maybe-uninitialized";
+  cmakeFlags = [ "-DCMAKE_POLKIT_PREFIX=${placeholder "out"}" ];
+
+  meta = with lib; {
+    description = "Analog Devices' oscilloscope software for Industrial IO (IIO) devices";
+    homepage = "https://wiki.analog.com/resources/tools-software/linux-software/iio_oscilloscope";
+    license = licenses.gpl2;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ii/iio-oscilloscope/package.nix:56]:
```suggestion
    license = licenses.free;
```
Fairly certain this isn't part of standard GPLv2
>  * - Neither the name of Analog Devices, Inc. nor the names of its
> *   contributors may be used to endorse or promote products derived
> *   from this software without specific prior written permission.

File: pkgs/by-name/ii/iio-oscilloscope/package.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  pcre2,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "iio-oscilloscope";
+  version = "0.17";
+
+  src = fetchFromGitHub {
+    owner = "analogdevicesinc";
+    repo = finalAttrs.pname;
```
Comment (FliegendeWurst) [inline pkgs/by-name/ii/iio-oscilloscope/package.nix:26]:
```suggestion
    repo = "iio-oscilloscope";
```
#277994

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by FliegendeWurst** [comment]
#280521 for the same package was merged.

**Pull request closed without merging**